src/main/java/com/qaprosoft/carina/demo/guipractice/components/NavigateMenu.java

```diff
@@ -14,36 +14,55 @@ public class NavigateMenu extends AbstractUIObject {
 	@FindBy(xpath = "//a[@id = 'login2']")
 	private ExtendedWebElement loginButton;
 
-	@FindBy(id = "//a[@id = 'signin2']")
+	@FindBy(xpath = "//a[@id = 'signin2']")
 	private ExtendedWebElement signinButton;
 
-	@FindBy(id = "//a[@id = 'logout2']")
+	@FindBy(xpath = "//a[@id = 'logout2']")
 	private ExtendedWebElement logoutButton;
 
-	@FindBy(id = "//a[@id = 'cartur']")
+	@FindBy(xpath = "//a[@id = 'cartur']")
 	private ExtendedWebElement cartButton;
 
+	@FindBy(xpath = "//a[@id = 'nameofuser']")
+	private ExtendedWebElement welcomeLabel;
+
 	public NavigateMenu(WebDriver driver) {
```

Comment:
Forgot to tell you this before: all components (classes that extends AbstractUIObject should have SearchContext in it
```
public NavigateMenu(WebDriver driver, SearchContext searchContext) {
		super(driver, searchContext);
	}
```
fix it for all your components